sympy/core/evalf.py

```diff
@@ -1528,6 +1526,8 @@ def evalf(self, n=15, subs=None, maxn=100, chop=False, strict=False, quad=None,
             except NotImplementedError:
                 # Probably contains symbols or unknown functions
                 return v
+        if self.doit() is S.ComplexInfinity: # issue 21147
+            return self.doit()
```

Comment:
These lines are calling `self.doit()` twice. It isn't computationally effecient.